api/src/main/java/io/opentelemetry/trace/WireFormatUtils.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace;
+
+public class WireFormatUtils {
```

Comment:
This is a sort of strange public API - not clear what wire format or version this is referring to, and pretty narrow use case. I think this was added because of the visibility of `BigEndianEncoding` but how about my suggestion of just precomputing the possible valid values and comparing to avoid needing this?

Reply:
set of 255 strings is not tiny. how I can compute all 255 values to insert into set? I suppose I will need to call `BigEndianEncoding.byteToBase16String(...)` method which is in `BigEndianEncoding` class.

Reply:
Well since this is on the slow path (static initializer), we don't quite need the performance of `BigEndianEncoding` and could just use `Long.toHexString`.

```
Set<String> VALID_VERSIONS = new HashSet<>();
for (int i = 0; i < 255; i++) {
  String version = Long.toHexString(i);
  if (version.length() < 2)
    version = '0' + version;
  VALID_VERSIONS.add(version);
}
```

It's not so bad I guess. We need to be careful with public API, for this I would probably copy rather than expose if we need to.

Reply:
I think the suggestion is to write some quick code to generate all the possibilities, and then just put that into a static Set field. Is that what you were thinking, @anuraaga ?

Reply:
done

Reply:
I added initialization to `static` block, can move to method if it's better

Reply:
The other option would be to pre-generate the set and hard-code it into the code itself, but I think this is fine a few extra microseconds at startup shouldn't be an issue, and we can change it to a hard-coded set later if we want.